Change imports (and other) from ubuntu-phonedations to ubports

jezek opened the pull request:
Follows PRs #6 & #7 

This PR proposes to cut ties from ubuntu-phonedations repo for good. This step is essential for further development of ```nuntium```. For example without this, any change to [nuntium/ofono package](https://github.com/ubports/nuntium/tree/xenial/ofono) wouldn't have any effect, cause ```nuntium``` [builds against ubutu-ponedations packages](https://github.com/ubports/nuntium/blob/38940464d532a145412c47d8901c1a74fb662eff/cmd/nuntium/main.go#L29-L30). I've splitted this changes into some commits for better clarity, with some insights in commit descriptions.

jezek committed:
Change imports from ubuntu-phonedations to ubports

jezek committed:
Change testing docs to use ubports, ...

Links to documentation don't work (yet), request to pkg.go.dev needs to
be made to make it work (which needs a license to be added to project).

Install on ubuntu statement was removed, cause it's not true anymore.

jezek committed:
Replace ubuntu-phonedations with ubports in debian package

I don't know, if I broke something with this commit, needs to be
investigate by someone who has experience with deb packaging.

jezek committed:
Add mention of ubuntu-phonedations to README.md

Enclose crossbuilder command as code

lduboeuf commented:
tested on BQ4.5, no regressions found and also the MMSEnabled toggle is working now. ( the MMS is not retrieved as expected when set to off , before it was not the case )

lduboeuf commented:
I would split this PR in two, one with the cherry-pick from phondation and the other for namespace migration ( just say that but i'm not the maintainer )

jezek commented:
@lduboeuf 
> I would split this PR in two, one with the cherry-pick from phondation and the other for namespace migration ( just say that but i'm not the maintainer )

Yep, it would be more logical. Maybe it's also fine this way. Will do, as the higher-ups say. :slightly_smiling_face:

jezek commented on cmd/nuntium/mediator.go:
```diff
@@ -421,3 +427,34 @@ func (mediator *Mediator) uploadFile(filePath string) (string, error) {
+		log.Printf("mmsEnabled: failed to get mms option reply: %v", err)
+		return true
+	}
+	log.Printf("mmsEnabled: mms option value: %v", mms.Value.(bool))
```
Now I know the code is working, the last log could be commented out. Just leave the logs after error.

dobey commented:
> Yep, it would be more logical. Maybe it's also fine this way. Will do, as the higher-ups say.

I think it would be better, or at least the first change should be to merge in the changes from the phonedations repo, by adding it as a remote and doing a `merge` or a `cherry-pick` rather than applying a diff, so that the commit history is properly preserved. The history should be compatible enough to do so, at least. Then all the imports and references to phonedations can be added separately, so there's a clear line of separation between the two.

Thanks!

dobey reviewed (changes requested):
See previous comment.

jezek commented:
@dobey You mean like this, or split the PR into two (mmsEnabled, imports)?

dobey commented:
> You mean like this, or split the PR into two (mmsEnabled, imports)?

Two separate PRs would probably be better. One for merging the "upstream" changes in (like maybe adding the remote and creating a new branch based on its master, and just PR that into xenial here), and a second with your changes. Or maybe three PRs since the "error logging and refactor mmsEnabled" bit is also separate from the replacing ubuntu-phonedations bit.

Thanks!

Flohack74 commented on README.md:
```diff
@@ -15,10 +15,11 @@ requirements that satisfy an Ubuntu Phone (aka Ubuntu Touch).
 
 * [mmsd documentaion](https://kernel.googlesource.com/pub/scm/network/ofono/mmsd/+/master/doc/)
-
+* This project is a continuitation of [the ubuntu-phonedations nuntium project](https://github.com/ubuntu-phonedations/nuntium/). 
```
Its _continuation_

jezek replied:
fixed

jezek commented:
@dobey 
> Two separate PRs would probably be better. One for merging the "upstream" changes in (like maybe adding the remote and creating a new branch based on its master, and just PR that into xenial here), and a second with your changes. Or maybe three PRs since the "error logging and refactor mmsEnabled" bit is also separate from the replacing ubuntu-phonedations bit.

To be sure, we are on the same page.

I've added upstream (ubuntu-phonedations/master) and then (after resolving some conflicts, due to new merged code in xenial) merged with xenial (see https://github.com/jezek/nuntium/tree/xenial_-_merge-upstream). This would be the PR no 1. 

Then the second would be error logging and refactoring of mms-enabled logic. 

An then the third would be this (without the first two commits and fixed reviews). Right?

dobey commented:
> Right?

Sounds good, yeah. Main concern was to preserve the upstream history. Thanks.

jezek commented:
@dobey Sorry for my delay. I've added PRs #6 & #7 , which should precede this one.

jezek commented:
Ok, now the PRs #6 & #7 are merged to xenial branch. What other changes need to be done to accept this PR?

dobey reviewed (approved):
Looks good. Thanks!
